homeassistant/components/swiss_public_transport/sensor.py

```diff
@@ -107,60 +95,29 @@ async def async_setup_platform(
         )
 
 
-class SwissPublicTransportSensor(SensorEntity):
+class SwissPublicTransportSensor(
+    CoordinatorEntity[SwissPublicTransportDataUpdateCoordinator], SensorEntity
+):
     """Implementation of an Swiss public transport sensor."""
 
     _attr_attribution = "Data provided by transport.opendata.ch"
     _attr_icon = "mdi:bus"
 
-    def __init__(self, opendata: OpendataTransport, name: str) -> None:
+    def __init__(
+        self, coordinator: SwissPublicTransportDataUpdateCoordinator, name: str
+    ) -> None:
         """Initialize the sensor."""
-        self._opendata = opendata
+        super().__init__(coordinator)
+        self._coordinator = coordinator
         self._attr_name = name
         self._remaining_time: timedelta | None = None
 
     @property
     def native_value(self) -> str:
         """Return the state of the sensor."""
-        return self._opendata.connections[0]["departure"]
+        return self.coordinator.data["departure"]
 
     @property
     def extra_state_attributes(self) -> Mapping[str, Any]:
         """Return the state attributes."""
-        departure_time = dt_util.parse_datetime(
-            self._opendata.connections[0]["departure"]
-        )
-        if departure_time:
-            remaining_time = departure_time - dt_util.as_local(dt_util.utcnow())
-        else:
-            remaining_time = None
-        self._remaining_time = remaining_time
-
-        return {
-            ATTR_TRAIN_NUMBER: self._opendata.connections[0]["number"],
-            ATTR_PLATFORM: self._opendata.connections[0]["platform"],
-            ATTR_TRANSFERS: self._opendata.connections[0]["transfers"],
-            ATTR_DURATION: self._opendata.connections[0]["duration"],
-            ATTR_DEPARTURE_TIME1: self._opendata.connections[1]["departure"],
-            ATTR_DEPARTURE_TIME2: self._opendata.connections[2]["departure"],
-            ATTR_START: self._opendata.from_name,
-            ATTR_TARGET: self._opendata.to_name,
-            ATTR_REMAINING_TIME: f"{self._remaining_time}",
-            ATTR_DELAY: self._opendata.connections[0]["delay"],
-        }
-
-    async def async_update(self) -> None:
-        """Get the latest data from opendata.ch and update the states."""
-
-        try:
-            if not self._remaining_time or self._remaining_time.total_seconds() < 0:
-                await self._opendata.async_get_data()
-        except OpendataTransportError:
-            self._attr_available = False
-            _LOGGER.warning(
-                "Unable to connect and retrieve data from transport.opendata.ch"
-            )
-        else:
-            if not self._attr_available:
-                self._attr_available = True
-                _LOGGER.info("Connection established with transport.opendata.ch")
+        return self.coordinator.data
```

Comment:
This actually now causes the state to become also an attribute.

Reply:
Yes, you are right. Good spot!
I filter it now on the sensor level fro the attributes